### DISCUSSION TITLE
Tutorial for Dialog

### DISCUSSION THREAD

**Discussion by Rafii2198** [tauri-apps/tauri#5331] [Q&A] [answered]
Hello! I am a beginner in Rust, I am working on an app for a specific game to combine many images related to that game. I finished making the image combining part itself, but I need the user to select the game folder from which the app will take the images. The thing is, I find working with the dialog a bit hard, probably I am confusing some stuff and the app is not working because of it. 
Can someone help me with it? For now, I want to make a button that will ask to select a folder and the path to that folder will be returned to the app as a variable. I think if I have this info, I can do the rest of the app.

**Comment by FabianLars** [comment, marked as answer]
javascript example: https://tauri.app/v1/api/js/dialog#open How it will work/look like in your app depends on your frontend stack.

There are a few examples in the rust docs too: <https://docs.rs/tauri/latest/tauri/api/dialog/blocking/struct.FileDialogBuilder.html#method.pick_file> just make sure to use the `blocking` dialog in an **async** command (the docs doesn't do that...) because the blocking one is way easier to use than the non-blocking one.
